dfusion_rust_core/src/models/account_state.rs

```diff
@@ -23,39 +24,64 @@ impl AccountState {
         AccountState {
             state_hash,
             state_index,
-            balances,
+            balances: balances
+                .chunks(num_tokens as usize)
+                .enumerate()
+                .map(|(account, token_balances)| {
+                    (
+                        U256::from(account),
+                        token_balances
+                            .iter()
+                            .enumerate()
+                            .map(|(token, balance)| (token as u8, *balance))
+                            .collect(),
+                    )
+                })
+                .collect(),
             num_tokens,
         }
     }
-    fn balance_index(&self, token_id: u8, account_id: u16) -> usize {
-        self.num_tokens as usize * account_id as usize + token_id as usize
+    pub fn get_balance_vector(&self) -> Vec<u128> {
+        let mut i = U256::zero();
+        let mut result = vec![];
+        while let Some(account) = self.balances.get(&i) {
```

Comment:
If you wanna introduce more syntax sugar, you might wanna implement the iterate trait for the balances and then simply iterate of it.
Not 100% sure, that this would work. But if it works, it would be more sugary.

Reply:
When iterating, I wouldn't be able to make sure that the ids are indeed adjacent. `{1: [0,0], 42, [0,0]}` would lead the same result as `{1: [0,0], 2, [0,0]}`, but shouldn't be a valid vector